File: src/main/java/de/tectoast/emolga/commands/pokemon/NatureCommand.java
package de.tectoast.emolga.commands.pokemon;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.sql.DBManagers;

import java.sql.SQLException;

public class NatureCommand extends Command {

    public NatureCommand() {
        super("nature", "Zeigt an, welche Werte dieses Wesen beeinflusst", CommandCategory.Pokemon);
        setArgumentTemplate(ArgumentManagerTemplate.builder().addEngl("nature", "Wesen", "Das Wesen", Translation.Type.NATURE)
                .setExample("!nature Adamant")
                .build());
    }

    @Override
    public void process(GuildCommandEvent e) throws SQLException {
        Translation t = e.getArguments().getTranslation("nature");
        e.reply(t.getOtherLang() + "/" + t.getTranslation() + ":\n" + DBManagers.NATURE.getNatureData(t.getTranslation()));
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/pokemon/NatureCommand.kt
package de.tectoast.emolga.commands.pokemon

import de.tectoast.emolga.commands.Command
import de.tectoast.emolga.commands.CommandCategory
import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.commands.Translation
import de.tectoast.emolga.database.exposed.NatureDB

class NatureCommand : Command("nature", "Zeigt an, welche Werte dieses Wesen beeinflusst", CommandCategory.Pokemon) {
    init {
        argumentTemplate =
            ArgumentManagerTemplate.builder().addEngl("nature", "Wesen", "Das Wesen", Translation.Type.NATURE)
                .setExample("!nature Adamant")
                .build()
    }

    override suspend fun process(e: GuildCommandEvent) {
        val t = e.arguments.getTranslation("nature")
        e.reply(
            "${t.otherLang}/${t.translation}:\n${NatureDB.getNatureData(t.translation)}"
        )
    }
}
